### DISCUSSION TITLE
Docker installation is confusing.

### DISCUSSION THREAD

**Discussion by Maximus1996** [0xERR0R/blocky#773] [General]
Hello,

Please forgive my ignorance, I admit I am new to selfhost and even newer to adblocking solutions.
I have to say that I find the installation guide for NAS (Synology) through Docker to be confusing. My main question would be, should all commands be run one after the other in that section? or is it one of them? Which means either use: "Docker from command line" OR "Run with docker-compose".

I know, for a newbie, I am probably better off using one of the "mainstream" Adblocking solutions (PiHole/Adguard) as they have more tutorials and support over forums and Reddit. But I don´t think I need all their fancy tools, I would like to use a simple solutions that works (fire and forget).

I would appreciate any hints. Thank you!

**Comment by kwitsch** [comment]
If you are new to the whole container world of Docker I would strongly advise using compose.
Preferable you install Docker with the "compose cli plugin" as it's the official implementation.

But I'm a little confused, the docker installation section consist only of one command.

**Reply by Maximus1996** [reply]
Thank you for the answer and the Tip! Yes I am new not only to Docker but selfhosting in general. 

So, in the installation page: https://0xerr0r.github.io/blocky/installation/
There are 4 sections:
Alternative registry
Parameters
Docker from Command line
Run with Docker-compose

I was not sure if "Docker from Command line" and "Run with Docker-compose" should be done one after the other or I needed to use just ONE of them. Now, I guess it is one of them which should be Docker-compose for me :) 
Thank you again!

**Comment by 0xERR0R** [comment]
I think Synology provides some kind of web UI to create a docker based app. Basically, you should create a config.yaml file (for example in a public share) and mount it as volume to "/app/config.yml". Also please expose ports 53 and 4000

**Reply by Maximus1996** [reply]
Thank you! I will look more into the configuration file itself and try to understand it better, I thought the exposition of the ports was only necessary for the use from "outside" the local network.

**Reply by 0xERR0R** [reply]
> Thank you! I will look more into the configuration file itself and try to understand it better, I thought the exposition of the ports was only necessary for the use from "outside" the local network.

I think, you need to expose the ports, otherwise you won't be able to access the DNS port from other device in your network

**Reply by ThinkChaos** [reply]
With docker, when you expose/publish a port, it means giving access to the port inside the container from the host, and other machines unless the host's firewall blocks that.  
This is separate from opening ports on your router, which I think is the confusion you had.  
See [`docker run`'s docs](https://docs.docker.com/engine/reference/commandline/run/#publish-or-expose-port--p---expose) and the [compose file reference](https://docs.docker.com/compose/compose-file/#ports) for more info.

**Reply by Maximus1996** [reply]
> With docker, when you expose/publish a port, it means giving access to the port inside the container from the host, and other machines unless the host's firewall blocks that. This is separate from opening ports on your router, which I think is the confusion you had. See [`docker run`'s docs](https://docs.docker.com/engine/reference/commandline/run/#publish-or-expose-port--p---expose) and the [compose file reference](https://docs.docker.com/compose/compose-file/#ports) for more info.

Ah ok, thank you! 
The project is very promising, I wish there were more tutorials/videos available for starting up. The only one I found was a Linux installation, which is good introduction, but different from NAS. 

Thank you for everyone involved in the project and I wish you more success :) !
